fmt::v8: bigint build failure with clang-10

```C++
#define FMT_HEADER_ONLY
#include <fmt/format.h>
int main() 
{
  fmt::detail::bigint b;
  b.square();
}
```

causes

```
In file included from /opt/compiler-explorer/libs/fmt/trunk/include/fmt/format.h:2983:
/opt/compiler-explorer/libs/fmt/trunk/include/fmt/format-inl.h:557:49: error: variable of non-literal type 'basic_memory_buffer<fmt::v8::detail::bigint::bigit, bigits_capacity>' (aka 'basic_memory_buffer<unsigned int, bigits_capacity>') cannot be defined in a constexpr function
    basic_memory_buffer<bigit, bigits_capacity> n(std::move(bigits_));
                                                ^
/opt/compiler-explorer/libs/fmt/trunk/include/fmt/format.h:651:13: note: 'basic_memory_buffer<unsigned int, 32, std::allocator<unsigned int> >' is not literal because it has data member 'alloc_' of non-literal type 'std::allocator<unsigned int>'
  Allocator alloc_;
            ^
```

repro: https://godbolt.org/z/nT1TP4Wof ; maybe FMT_CONSTEXPR20 needs to start from clang-11 (where the above code compiles without issues)

--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -94,7 +94,8 @@
 #  define FMT_CONSTEXPR_DECL
 #endif
 
-#if __cplusplus >= 202002L || \
+#if ((__cplusplus >= 202002L) && \
+     (!defined(_GLIBCXX_RELEASE) || _GLIBCXX_RELEASE > 9)) || \
     (__cplusplus >= 201709L && FMT_GCC_VERSION >= 1002)
 #  define FMT_CONSTEXPR20 constexpr
 #else

